src/h5web/vis-packs/core/matrix/MappedMatrixVis.tsx

```diff
@@ -17,9 +18,18 @@ interface Props {
 function MappedMatrixVis(props: Props): ReactElement {
   const { dataset, dims, dimMapping } = props;
 
-  const value = useDatasetValue(dataset);
-  const baseArray = useBaseArray(value, dims);
-  const mappedArray = useMappedArray(baseArray, dimMapping);
+  const value = useDatasetValue(dataset, dimMapping);
+
+  const [slicedDims, slicedMapping] = useMemo(
+    () => [
+      dimMapping.flatMap((dim, i) => (isAxis(dim) ? [dims[i]] : [])),
+      dimMapping.filter((dim) => isAxis(dim)),
+    ],
+    [dimMapping, dims]
+  );
+
+  const baseArray = useBaseArray(value, slicedDims);
+  const mappedArray = useMappedArray(baseArray, slicedMapping);
```

Comment:
Same code as in `MappedHeatmapVis`. Wasn't sure it was worth/a good idea to abstract it or to rethink this whole section of code. I don't mind the duplication too much, but if you have a better idea 🤷 

I can't just pass `dimMapping` and `dims` to `useBaseArray` and compute `slicedDims` in there, because I don't want to compute `slicedDims` in `MappedLineVis`... Same goes with `useMappedArray`...

Reply:
The `useMemo` computing `slicedDims` and `slicedMapping` could be put in a shared hook. 

The rest is fine as it is bound to change anyway.

Reply:
I'll do it in the next PR, in case this is impacted by the refactoring of the hooks.